File: app/src/main/java/com/app/scavenger/About.java
package com.app.scavenger;

import androidx.appcompat.app.AppCompatActivity;
import androidx.browser.customtabs.CustomTabsIntent;
import androidx.core.content.ContextCompat;
import androidx.preference.PreferenceManager;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.content.ActivityNotFoundException;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.widget.ImageButton;
import java.util.ArrayList;

// About activity that lists information about Scavenger

public class About extends AppCompatActivity implements AboutAdapter.ItemClickListener {

    private boolean inAppBrowsingOn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);

        RecyclerView aboutRecycler = findViewById(R.id.about_list);

        TopToolbar topToolbar = findViewById(R.id.about_toolbar);
        topToolbar.setTitle("About");

        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);

        // Add options inside of the recyclerview
        ArrayList<String> options = new ArrayList<>();
        options.add(getResources().getString(R.string.terms_and_conditions));
        options.add(getResources().getString(R.string.privacy_policy));
        options.add(getResources().getString(R.string.open_source_libraries));

        //aboutRecycler.setLayoutManager(new LinearLayoutManager(this));
        AboutAdapter adapter = new AboutAdapter(this, options);
        adapter.setClickListener(this);
        aboutRecycler.setAdapter(adapter);

        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(this, layoutManager.getOrientation());
        aboutRecycler.addItemDecoration(dividerItemDecoration);
        aboutRecycler.setLayoutManager(layoutManager);

        inAppBrowsingOn = sharedPreferences.getBoolean("inAppBrowser", true);


        // Close activity through back button
        //backButton.setOnClickListener(v -> finish());

    }

    @Override
    public void onItemClick(int position) {

        switch (position) {
            // Terms and Conditions
            case 0:
                if (inAppBrowsingOn) {
                    openURLInChromeCustomTab(this, Constants.scavengerTermsURL);
                } else {
                    openInDefaultBrowser(this, Constants.scavengerTermsURL);
                }
                break;
                // Privacy Policy
            case 1:
                if (inAppBrowsingOn) {
                    openURLInChromeCustomTab(this, Constants.scavengerPrivacyURL);
                } else {
                    openInDefaultBrowser(this, Constants.scavengerPrivacyURL);
                }
                break;
                // Open Source Libraries
            case 2:
                openOSL();
                break;
        }
    }

    // Method for opening the Open Source Libraries activity
    private void openOSL() {
        Intent intent = new Intent(this, OpenSourceLibraries.class);
        startActivity(intent);
    }

    // opens the recipe in the users default browser
    private void openURLInChromeCustomTab(Context context, String url) {
        try {
            CustomTabsIntent.Builder builder1 = new CustomTabsIntent.Builder();
            CustomTabsIntent customTabsIntent = builder1.build();
            customTabsIntent.intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            builder1.setInstantAppsEnabled(true);
            customTabsIntent.intent.putExtra(Intent.EXTRA_REFERRER, Uri.parse("android-app://" + context.getPackageName()));
            builder1.setToolbarColor(ContextCompat.getColor(context, R.color.colorPrimaryDark));
            customTabsIntent.launchUrl(context, Uri.parse(url));
        } catch (ActivityNotFoundException e) {
            e.printStackTrace();
            Log.e("ChromeCustomTabError: ", "Activity Error");
        }
    }

    // open the recipe in the App Browser
    private void openInDefaultBrowser(Context context, String url) {
        try {
            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
            context.startActivity(browserIntent);
        } catch (ActivityNotFoundException e) {
            e.printStackTrace();
            Log.e("DefaultBrowserError: ", "Activity Error");
        }
    }
}

File: app/src/main/java/com/app/scavenger/v2/activities/About.kt
package com.app.scavenger.v2.activities

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.app.scavenger.AboutAdapter
import com.app.scavenger.utils.Constants
import com.app.scavenger.OpenSourceLibraries
import com.app.scavenger.R
import com.app.scavenger.databinding.ActivityAboutBinding

class About: AppCompatActivity(), AboutAdapter.ItemClickListener {

    private lateinit var binding: ActivityAboutBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityAboutBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.aboutToolbar.setTitle("About")

        val options = ArrayList<String>()
        options.add(getString(R.string.terms_and_conditions))
        options.add(getString(R.string.privacy_policy))
        options.add(getString(R.string.open_source_libraries))
        options.add(getString(R.string.scavenger_base_url))

        val aboutAdapter = AboutAdapter(this, options)
        aboutAdapter.setClickListener(this)
        binding.aboutList.adapter = aboutAdapter
        binding.aboutList.addItemDecoration(DividerItemDecoration(this, RecyclerView.VERTICAL))
        binding.aboutList.layoutManager = LinearLayoutManager(this)
    }

    override fun onItemClick(position: Int) {

        when (position) {
            0 -> open(Constants.scavengerTermsURL)
            1 -> open(Constants.scavengerPrivacyURL)
            2 -> openOSL()
            3 -> open(Constants.scavengerBaseURL)
            else -> open(Constants.scavengerBaseURL)
        }
    }

    private fun openOSL() {
        val intent = Intent(this, OpenSourceLibraries::class.java)
        startActivity(intent)
    }

    private fun open(url: String) {
        try {
            val intent = Intent(Intent.ACTION_VIEW)
            intent.data = Uri.parse(url)
            startActivity(intent)
        } catch (e : ActivityNotFoundException) {
            e.printStackTrace()
        }
    }
}